design/one-pager-api-versioning-policy.md

```diff
@@ -0,0 +1,393 @@
+release**
+
+**Deprecation Implications:**
+
+- Version will not receive new features
+- Only bug fixes or conversion-related patches accepted
+- Removal may occur after one or more stable cycles, depending on provider
+maturity
+
+### Version Removal
+
+Removing an API version constitutes a **breaking change** by definition.
+However, given that major version bumps are infrequent for providers, controlled
+removals may occur during a minor version release if **all** of the following
+are true:
+
+1. The deprecated version has been retained for at least one prior minor release
+2. The newer storage version has been verified stable in production
+3. Conversion logic is validated to ensure safe object reads from the deprecated
+version
+4. The removal is explicitly noted as a breaking change in release documentation
+
+**Important Note:**
+
+Setting `served=false` for an API version is a breaking change for any client
+still issuing requests using that version. Even though conversion logic
+continues to operate internally, external consumers lose the ability to interact
+with that version via the Kubernetes API. For this reason, `served=false`
+transitions must be documented under "Breaking Changes" in release notes.
+
+### Sequential Upgrade and Downgrade Requirements
+
+Upjet's bridge-version model introduces a **sequential upgrade and downgrade
+requirement**:
+
+- **Upgrades must follow N → N+1 ordering**
+- **Downgrades must follow N → N-1 ordering**
+- **Skipping versions is not supported**
+
+Each release may introduce a new storage schema or serve/don't-serve
+configuration that older versions cannot interpret.
+
+This aligns with established Kubernetes ecosystem best practices:
+- [Kubernetes kubeadm upgrade guidelines](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade)
+- [Amazon EKS in-place upgrade best practices](https://docs.aws.amazon.com/eks/latest/best-practices/cluster-upgrades.html#upgrade-clusters-in-place)
+
+**Downgrade Compatibility:**
+
+Per this policy, downgrade compatibility is only guaranteed across
+**consecutive** provider releases (N → N-1), for example v1.2.0 → v1.1.0. Direct
+downgrades that skip an intermediate release (e.g., v1.2.0 → v1.0.0) are **not
+supported**.
+
+This constraint follows directly from the bridge-version model: once etcd begins
+to persist objects under a newer storage version schema, older releases unaware
+of that schema cannot reliably read or reconcile those objects.
+
+This design favors forward stability while providing a safe, one-step downgrade
+path for recovery scenarios, at the cost of disallowing multi-step downgrades.
+
+### Lifecycle Summary
+
+1. **Introduce:** Add a new API version alongside the current one
+2. **Bridge:** Maintain both versions, keeping the old one as storage
+3. **Deprecate:** Promote the new version to storage and mark the old one
+deprecated
+4. **Block:** Set `served=false` to block (REST) clients from using the version
```

Comment:
is this something we can show people via metrics - that still REST clients asking for the `served=false` version ?

Reply:
This can be verified through further investigation/consideration and added to a future work item for adding as a new upjet metric.